### DISCUSSION TITLE
How to prevent plugin from compilling js?

### DISCUSSION THREAD

**Discussion by JONMARSHALDEV** [webdiscus/pug-plugin#24] [General]
i cant figure how to prevent plugin from compilling js.
i want to include "js" as-is.
for example i have script:
`
console.log('\n>> Home page');
var a = "MARSHALL";
console.log(a);
var b = a + " JOHN";
console.log(b);`

it includes, but webpack processing it to:
`(()=>{console.log("\n>> Home page");console.log("MARSHALL"),console.log("MARSHALL JOHN")})();`

i'm adding script to head:
`
append head
  script(src=require('./main.js') defer='')
`

pls help.

**Comment by webdiscus** [comment]
@JONMARSHALDEV 

The Pug plugin doesn't play any role here.

The native Pug syntax to include raw JavaScript source in resulting HTML:

```pug
append head
  script: include ./main.js
```

If you need to load raw JS as external file, then place this file to public directory with compiled js files, e.g. `/assests/js/main.js` and write in Pug:

```pug
append head script(src='/assets/js/main.js' defer='')
```

To copy your raw source file in public directory, you can use the ['copy-webpack-plugin'](https://webpack.js.org/plugins/copy-webpack-plugin/)

Otherwise is impossible to prevent the Webpack from compilling js.

**Reply by JONMARSHALDEV** [reply]
ok. tnx.
